### PR TITLE
Reinstated save_png method of figure

### PR DESCRIPTION
**Pull request by dmadeka**
@SylvainCorlay Brought back the save_png function for the `Figure`, @stonebig asked for a stable beta, I think this should probably be included

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
The issue is that the save_png makes each view save the png.

**Comment by dmadeka** [comment]
As in all views or the current view? Because the latter seems to be what we want right?

----- Original Message -----
From: Sylvain Corlay <reply@reply.github.com>
To: bqplot@noreply.github.com
CC: DHRUV MADEKA, author@noreply.github.com
At: 17-Jun-2016 15:42:08


The issue is that the save_png makes each view save the png.

**Comment by SylvainCorlay** [comment]
All the views... It is a kernel function that sends a message to all the
views...

That is why we need to make the toolbar a child widget and have the save
button only save the one of the view of the parent figure...

**Comment by dmadeka** [comment]
Alternately, we could have a little button on the `Figure` itself right? A controllable mini-toolbar thats displayed or not and clicking that could save the `Figure`. Because right now to save, you have to create a `Toolbar` instance, plug it to the `Figure` and then display it and click save. There's no Pythonic way or simple way to do it

**Comment by SylvainCorlay** [comment]
Actually, the direction I wanted to take was to move up the toolbar widget
as a child widget and have the figure instantiate a default toolbar. This
would make it more easily extensible ...

Sylvain

**Comment by dmadeka** [comment]
Well. Ideally, the `Toolbar` should allow you to configure any properties of the `Figure` right? So that's the right way to go I guess

**Comment by dmadeka** [comment]
btw @SylvainCorlay will the `Toolbar` as a child solve the issue that we aren't able to programatically save a `png`?

**Comment by SylvainCorlay** [comment]
No, I am still ok with your PR!

**Comment by dmadeka** [comment]
okay @SylvainCorlay yours to merge then

File: js/src/Figure.js [hunk 1]
```diff
@@ -173,10 +173,18 @@ define(["jupyter-js-widgets", "d3", "underscore"],
...
+				  that);
                 });
             });
         },
 
+	handle_custom_messages: function(msg) {
+            if (msg.type === 'save_png') {
+		    this.save_png();
+	    }
+	},
```
Comment (SylvainCorlay) [inline js/src/Figure.js]: ah, formatting.

**Comment by SylvainCorlay** [comment]
Thanks

**Pull request merged**